terraform/eks/daemon/fluent/windows/main.tf

```diff
@@ -106,33 +157,365 @@ resource "aws_eks_node_group" "node_group_windows" {
+  source_security_group_id = aws_security_group.eks_cluster_sg.id
+  to_port                  = 65535
+  type                     = "ingress"
+}
+
+resource "kubernetes_namespace" "namespace" {
+  metadata {
+    name = "amazon-cloudwatch"
+  }
+}
+
+resource "kubernetes_service_account" "cwagentservice" {
+  depends_on = [kubernetes_namespace.namespace]
+  metadata {
+    name      = "cloudwatch-agent"
+    namespace = "amazon-cloudwatch"
+  }
+}
+
+resource "kubernetes_cluster_role" "clusterrole" {
+  depends_on = [kubernetes_namespace.namespace]
+  metadata {
+    name = "cloudwatch-agent-role"
+  }
+  rule {
+    verbs      = ["list", "watch"]
+    resources  = ["pods", "nodes", "endpoints"]
+    api_groups = [""]
+  }
+  rule {
+    verbs      = ["list", "watch"]
+    resources  = ["replicasets"]
+    api_groups = ["apps"]
+  }
+  rule {
+    verbs      = ["list", "watch"]
+    resources  = ["jobs"]
+    api_groups = ["batch"]
+  }
+  rule {
+    verbs      = ["get"]
+    resources  = ["nodes/proxy"]
+    api_groups = ["get"]
+  }
+  rule {
+    verbs      = ["create", "get"]
+    resources  = ["nodes/stats", "configmaps", "events"]
+    api_groups = [""]
+  }
+  rule {
+    verbs          = ["get", "update"]
+    resource_names = ["cwagent-clusterleader"]
+    resources      = ["configmaps"]
+    api_groups     = [""]
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "rolebinding" {
+  depends_on = [
+    kubernetes_service_account.cwagentservice,
+    kubernetes_cluster_role.clusterrole
+  ]
+  metadata {
+    name = "cloudwatch-agent-role-binding"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "cloudwatch-agent-role"
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = "cloudwatch-agent"
+    namespace = "amazon-cloudwatch"
+  }
+}
+
+resource "kubernetes_config_map" "cwagentconfig" {
+  metadata {
+    name      = "cwagentconfig"
+    namespace = "amazon-cloudwatch"
+  }
+  data = {
+    "cwagentconfig.json" = <<EOF
+    {
+      "agent": {
+        "region": "${var.region}"
+      },
+      "logs": {
+        "metrics_collected": {
+          "kubernetes": {
+            "cluster_name": "${aws_eks_cluster.cluster.name}",
+            "metrics_collection_interval": 60
+          }
+        },
+        "force_flush_interval": 5
+      }
+    }
+    EOF
+  }
+}
+
+resource "kubernetes_daemonset" "agent_daemon" {
+  depends_on = [
+    kubernetes_config_map.cwagentconfig
+  ]
+  metadata {
+    name      = "cloudwatch-agent"
+    namespace = "amazon-cloudwatch"
+  }
+  spec {
+    selector {
+      match_labels = {
+        "name" : "cloudwatch-agent"
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          "name" : "cloudwatch-agent"
+        }
+      }
+      spec {
+        container {
+          name  = "cloudwatch-agent"
+          image = "public.ecr.aws/cloudwatch-agent/cloudwatch-agent:1.247359.1b252618"
```

Comment:
why is an older version of the agent being used ?

Reply:
This logic is similar to fluent bit test on Linux.
Since this is fluent bit test, CW agent version shouldn't matter.

Reply:
agree that agent version doesn't really matter in the context of fluentbit/d tests, but it would be nice to update it while we are at it. I think it's also a good practice to not keep a legacy agent version in Github. 

Can you please update it to `1.300040.0b650`? The other file that has this old version is `terraform/eks/daemon/fluent/common/main.tf`.

Reply:
Addressed.
Update Cw agent version to `1.300040.0b650` at both the places.